### PR TITLE
TWM-56: Update test matrix.

### PR DESCRIPTION
**Pull request by adam-vessey**
No D9, PHP 7.4, start introducing the different D10 minor versions.

### CODE REVIEW

File: .github/workflows/phpunit.yml [hunk 1]
```diff
@@ -20,29 +20,35 @@ jobs:
...
+            php-version: '8.3'
             # XXX: experimental, for now.
             experimental: true
             do-leniency: true
+            patches: |-
+              {
+                "drupal/core": {
+                  "Referencable unpublished nodes": "https://www.drupal.org/files/issues/2024-01-07/2845144-78.patch"
+                }
+              }
```
Comment (adam-vessey) [inline .github/workflows/phpunit.yml:50]: Found that https://github.com/discoverygarden/phpunit-action/pull/19/files#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6R172-R174 was failing on D10.2, so slapped together a mechanism so different versions can pass their own patches... is kinda gross, but whatever.

File: .github/workflows/phpunit.yml [hunk 2]
```diff
@@ -68,7 +74,8 @@ jobs:
           apt-get install -y postgresql-client
           PGHOST=postgres PGUSER=drupal PGPASSWORD=drupal psql --command='CREATE EXTENSION pg_trgm;' drupal
       - name: Run PHPUnit Action
-        uses: discoverygarden/phpunit-action@v1
+        #uses: discoverygarden/phpunit-action@v1
+        uses: discoverygarden/phpunit-action@adjust-matrix
```
Comment (JojoVes) [inline .github/workflows/phpunit.yml:78]: Remember to adjust before merging

Comment (nchiasson-dgi) [reply .github/workflows/phpunit.yml]: Adjusted back to the major version tag v1

**Comment by nchiasson-dgi** [comment]
Tested and reviewed within the provided module. Everything looks as it should. Merging this in.

**Pull request merged**